src/launcher/java/org/truffleruby/launcher/Launcher.java

```diff
@@ -204,29 +204,42 @@ public static void processArguments(
             List<String> args,
             boolean parseHelpEtc,
             boolean unknownOptionFails,
-            boolean allowedJVMOptions) {
+            boolean isNative) {
 
         try {
             config.setOption(OptionsCatalog.EXECUTION_ACTION, ExecutionAction.UNSET);
 
-            new CommandLineParser(args, config, true, false, parseHelpEtc).processArguments();
+            final CommandLineParser argumentCommandLineParser = new CommandLineParser(args, config, true, false, parseHelpEtc);
+            argumentCommandLineParser.processArguments();
 
             if (unknownOptionFails && !config.getUnknownArguments().isEmpty()) {
                 throw new CommandLineException("unknown option " + config.getUnknownArguments().get(0));
             }
 
             if (config.getOption(OptionsCatalog.READ_RUBYOPT)) {
-                new CommandLineParser(getArgsFromEnvVariable("RUBYOPT"),
-                        config, false, true, true).processArguments();
-                new CommandLineParser(getArgsFromEnvVariable("TRUFFLERUBYOPT"),
-                        config, false, false, true).processArguments();
+                final List<String> rubyoptArgs = getArgsFromEnvVariable("RUBYOPT");
+                final List<String> trufflerubyoptArgs = getArgsFromEnvVariable("TRUFFLERUBYOPT");
+                new CommandLineParser(rubyoptArgs, config, false, true, true).processArguments();
+                new CommandLineParser(trufflerubyoptArgs, config, false, false, true).processArguments();
+
+                if (isNative) {
```

Comment:
Why is this done only if `isNative`?

Reply:
see updated comment.

Reply:
> Only native launcher can apply native and jvm options, therefore this is not done on JVM.

So in a `--jvm` mode we parse arguments twice because they are first parsed and transformed by the native launcher; and the launcher parsing is executed again on JVM?
But then, those JVM arguments should already be consumed when we are in parsing again on JVM, right?
Can you confirm that's correct?
If so, `isNative` is not very explicit, maybe `isFirstLauncher` or so.

Reply:
That's correct, that is how the `AbstractLanguageLauncher` works. `isNative` is used everywhere in related code, I'll rather stick to it.

Reply:
OK